File: swift/problem191.swift
//ref: https://leetcode.com/problems/number-of-1-bits/solution/
class Solution {
    func hammingWeight(_ n: UInt32) -> Int {
        var bitCount = 0; var n = n
        while (n != 0) {
            bitCount += 1
            n &= (n-1)
        }
        return bitCount
    }
}

File: python/problem191.py
class Solution(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        nn = '0'*(34-len(bin(n))) + bin(n)[2::]
        
        nu=0
        for i in nn:
            if int(i) == 1:
                nu+=1
        return nu
#    https://leetcode.com/problems/number-of-1-bits/discuss/55106/Python-2-solutions.-One-naive-solution-with-built-in-functions.-One-trick-with-bit-operation    
class Solution2(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        c = 0
        while n:
            n &= n - 1
            c += 1
        return c